libcudacxx/include/cuda/__ptx/instructions/mul.h

```diff
@@ -0,0 +1,160 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA_PTX_MUL_H_
+#define _CUDA_PTX_MUL_H_
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/__ptx/ptx_dot_variants.h>
+#include <cuda/std/cstdint>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA_PTX
+
+_CCCL_DEVICE static inline _CUDA_VSTD::int16_t
```

Comment:
`_CUDA_VSTD` should be used only for functions (ADL)

Reply:
The problem is that the `int16_t` is in `_CUDA_VSTD` namespace. Other already implemented PTX instructions use the way of refering types in `cuda::std` namespace.

Reply:
sorry, you are right. I got confused about the usage of `_CUDA_VSTD` outside `cuda/std` headers. It should be `::cuda::std` instead of `_CUDA_VSTD`, but it makes more sense to be consistent with other PTX code